crates/bevy_input/src/gamepad.rs

```diff
@@ -108,12 +107,28 @@ impl Gamepad {
     reflect(Serialize, Deserialize)
 )]
 pub struct GamepadInfo {
-    /// The name of the gamepad.
+    /// The display name of the gamepad.
     ///
-    /// This name is generally defined by the OS.
+    /// This name is generally defined by the OS. For example on Windows the
+    /// name may be "HID-compliant game controller". The name is not unique,
+    /// typically two gamepads of the same model will have the same name.
+    /// For a unique identifier stable across runs, see [`uuid`] instead.
     ///
-    /// For example on Windows the name may be "HID-compliant game controller".
+    /// [`uuid`]: GamepadInfo::uuid
     pub name: String,
+
+    /// The unique identifier of the gamepad.
+    ///
+    /// This UUID is computed by `gilrs` and is guaranteed to be unique for each
+    /// device, even if they're the same make and model. The ID is also stable
+    /// across runs, so can be used to identify a gamepad in serialized data for
+    /// example.
+    ///
+    /// # Warning
+    ///
+    /// The value is extracted from `gilrs`. At the moment, this is not
+    /// implemented on all platforms. See <https://gitlab.com/gilrs-project/gilrs/-/issues/153>.
+    pub uuid: Uuid,
```

Comment:
Until it's not implemented on all (or at least most) platforms, wouldn't it be better to store it as `Option<Uuid>`? We could check if `uuid.is_nil()`, and save it accordingly. This would give better control for developers, to handle possible edge cases. Also that way it would be self-documenting that it might not be available.

Reply:
I thought about this too. But I'm worried about the increased burden on developers consuming such an API.

As a developer using the API, what do I do if the API doesn't guarantee a UUID is always available? The expectation is that the UUID must be unique, but now it may not be because it may not exist. In short, it's not unique. And in all use cases I can think of, the answer is "I don't use that API in the first place, because that would break my feature".

I'd much rather have a comment saying "this is only available on platform X and Y", which leaves the user choose if they want to take that risk. `Option<Uuid>` doesn't give any guarantee that two consecutive runs of the same app on the same device will both return a UUID. And if I'm going to `.unwrap()` because I know the docs said this is implemented on the platform I use, then I'm just working around the API.